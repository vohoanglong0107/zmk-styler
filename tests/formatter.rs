use std::fs;

use zmk_styler::format;

#[test]
fn test_formatter() {
    let mut settings = insta::Settings::clone_current();
    settings.set_snapshot_path(".");
    let in_file = "tests/glove80.keymap";
    let out_file = "tests/formatted.keymap";
    format(in_file, out_file).unwrap();
    let out = fs::read_to_string(out_file).unwrap();
    settings.bind(|| insta::assert_snapshot!("formatted", out))
}
