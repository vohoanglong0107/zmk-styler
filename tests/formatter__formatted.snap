---
source: tests/formatter.rs
expression: out
---
/ {
    behaviors {
        lower: lower {
            compatible = "zmk,behavior-tap-dance";
            label = "LAYER_TAP_DANCE";
            tapping-term-ms = <200>;
        };
    };
};
